### ISSUE TITLE
Ad hoc reconnection

### ISSUE THREAD

**Issue by njoi** [shtejv/ARDrone-Control-.NET#24]
If you use the Free.Flight AR.Drone application, if you leave the ad hoc network it will pop up saying video not available etc, and then if u move back into the ad hoc network range it will re-connect to the drone and allow you to control it again.  

with this software, the ARDrone Control .NET application, if the ad hoc network is left OR has a small hiccup in connectivity, it throws a pretty hefty socket error at the user and the application has to be restarted once manually reconnected to the network. Is there any plans to improve this networking issue or am I on my own here? I am using the application to do some testing with and I would like to be able to reconnect via better means than restarting all the applications and network connections.  it also appears that even if the network doesn't disconnect but possibly has a lag or hiccup it will also throw an error that requires full network reconnection and application restarting...

**Comment by shtejv** [comment]
Yes, the network code is still a bit troublesome. The NetworkWorker classes can be rewritten to change this behaviour.

**Comment by AndersMalmgren** [comment]
I too have noticed this, this problem makes any program written with the API too unstable to use outdoors.
I've looked at the API code, but its alot of code to go through, I hope you guys take alook at this for the next version because its easily the most important issue right now. Love your work guys!

**Comment by njoi** [comment]
I've spent a few days trying to get this to work but haven't had any luck. The error that I am repeatedly hitting and have found absolutely no workaround to is:

An exception 'System.Net.Sockets.SocketException' occured:
Only one usage of each socket address (protocol/network address/port) is normally permitted

This will occur any time I try to reconnect to the Drone, my computer IS connected to the drone network I've checked a bunch of different settings and tried a few different ways of reconnecting and nothing seems to successfully avoid this error... Any help would be greatly appreciated

**Comment by shtejv** [comment]
Correct, these are problematic issues ... But at the time, I (Thomas) am the only one left in the development 'team'. So help would really be appreciated.
